From d28754fa01274b129521260dff14a44db62beb40 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Fri, 31 Mar 2017 13:48:48 +0800
Subject: [PATCH 45/62] text: add user to sudo group as administrator

Add user to group 'sudo' rather than 'wheel' as administrator.

Upstream-Status: Inappropriate [wrlinux specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 pyanaconda/ui/tui/spokes/password.py |  2 +-
 pyanaconda/ui/tui/spokes/user.py     | 14 +++++++-------
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/pyanaconda/ui/tui/spokes/password.py b/pyanaconda/ui/tui/spokes/password.py
index fea1f7e..5013ba2 100644
--- a/pyanaconda/ui/tui/spokes/password.py
+++ b/pyanaconda/ui/tui/spokes/password.py
@@ -49,7 +49,7 @@ class PasswordSpoke(FirstbootSpokeMixIn, EditTUIDialog):
     @property
     def mandatory(self):
         return not any(user for user in self.data.user.userList
-                       if "wheel" in user.groups)
+                       if "sudo" in user.groups)
 
     @property
     def status(self):
diff --git a/pyanaconda/ui/tui/spokes/user.py b/pyanaconda/ui/tui/spokes/user.py
index ade78a8..ac851a2 100644
--- a/pyanaconda/ui/tui/spokes/user.py
+++ b/pyanaconda/ui/tui/spokes/user.py
@@ -86,7 +86,7 @@ class UserSpoke(FirstbootSpokeMixIn, EditTUISpoke):
         self.initialize_done()
 
     def refresh(self, args = None):
-        self.args._admin = "wheel" in self.args.groups
+        self.args._admin = "sudo" in self.args.groups
         self.args._groups = ", ".join(self.args.groups)
 
         # if we have any errors, display them
@@ -124,7 +124,7 @@ class UserSpoke(FirstbootSpokeMixIn, EditTUISpoke):
             return _("No user will be created")
         elif self.args._use_password and not bool(self.args.password or self.args.isCrypted):
             return _("You must set a password")
-        elif "wheel" in self.data.user.userList[0].groups:
+        elif "sudo" in self.data.user.userList[0].groups:
             return _("Administrator %s will be created") % self.data.user.userList[0].name
         else:
             return _("User %s will be created") % self.data.user.userList[0].name
@@ -159,11 +159,11 @@ class UserSpoke(FirstbootSpokeMixIn, EditTUISpoke):
 
         self.args.groups = [g.strip() for g in self.args._groups.split(",") if g]
 
-        # Add or remove the user from wheel group
-        if self.args._admin and "wheel" not in self.args.groups:
-            self.args.groups.append("wheel")
-        elif not self.args._admin and "wheel" in self.args.groups:
-            self.args.groups.remove("wheel")
+        # Add or remove the user from sudo group
+        if self.args._admin and "sudo" not in self.args.groups:
+            self.args.groups.append("sudo")
+        elif not self.args._admin and "sudo" in self.args.groups:
+            self.args.groups.remove("sudo")
 
         # Add or remove the user from userlist as needed
         if self.args._create and (self.args not in self.data.user.userList and self.args.name):
-- 
2.7.4

